android/src/main/kotlin/com/onegini/mobile/sdk/flutter/providers/CustomTwoStepRegistrationActionImpl.kt

```diff
@@ -39,27 +37,29 @@ class CustomTwoStepRegistrationActionImpl(private val providerId: String) : Oneg
         return providerId
     }
 
-    override fun returnSuccess(result: String?, pigeonChannel: (Result<Unit>) -> Unit) {
-        when (callback) {
-            null -> pigeonChannel(Result.failure(SdkError(OneWelcomeWrapperErrors.REGISTRATION_NOT_IN_PROGRESS).pigeonError()))
+    override fun returnSuccess(result: String?): Result<Unit> {
```

Comment:
You could do something like this instead for all these cases 
```
        return callback?.let { customRegistrationCallback ->
            customRegistrationCallback.returnError(exception)
            callback = null
            Result.success(Unit)
        } ?: Result.failure(SdkError(REGISTRATION_NOT_IN_PROGRESS).pigeonError())
```

Note here that the last line in the body of the .let is returned, this is also clearly indicated in Android Studio when applying this pattern.